src/main/java/spoon/support/compiler/jdt/TreeBuilderCompiler.java

```diff
@@ -72,6 +72,8 @@
 			requestor.acceptResult(unit.compilationResult);
 		}
 
-		return this.unitsToProcess;
+		CompilationUnitDeclaration[] unitsToReturn = new CompilationUnitDeclaration[this.totalUnits];
+		System.arraycopy(this.unitsToProcess, 0, unitsToReturn, 0, this.totalUnits);
```

Comment:
> I can change it to scan the array and only copy non-null values
That would be better for future maintenance, thanks.

You can also document this contract in the Javadoc (returns an array of
non-null elements).

--Martin